src/Source/Source.js

```
@@ -0,0 +1,31 @@
+class Source {
+    /**
+     * @param {string[]} tags  The tags to respond to
+     */
+    constructor(tags) {
+        this.tags = tags;
+    }
+
+    /**
+     * Check if the source has any of the given tags
+     * @param {string[]} tags The tags we're looking for
+     * @return {boolean}      Whether one of the tags corresponds to our own
+     */
+    isRelevantToAnyOfThese(tags) {
+        for (let tag in this.tags) {
```

I'd put `throw 'Method not defined'` here instead tbh

---

Although, I guess it'd probably be the same for all of them